### ISSUE TITLE
Gist hits GitHub API rate limits quickly

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang/rust-playpen#138]
Unfortunately without authenticated API requests we're rate-limited to 60 API requests per hour from any one IP, and if you're at a conference, for example, you hit this fairly quickly!

The best solution here is probably:

* Send gist upload/download requests to the play server
* Use OAuth authentication to get a higher rate limit on the play server.

cc @pnkfelix, cc @whipsch

**Comment by whipsch** [comment]
Oops! I never thought of conferences/other gatherings on shared internet access, I just assumed one user most likely wouldn't view/publish more than 1 Gist per minute. Guess not!

Authenticating should be done, additionally so because it allows for more Gist features to be utilized.  Forking and revising via the playpen as your own user would be nice to have!  As far as I can tell, anonymous user's cant edit their Gists (or delete them), though.

While I don't think the playpen would use more than 5000 requests per hour (Maybe in the future? Do we have stats for this?), it may be nice to default to using Gist anonymously and then resort to Gisting through the server if the 60/hr rate limit has been hit.  The user can still authenticate initially if they want to, of course.

**Comment by alexcrichton** [comment]
Yeah I think that 5k/hr is fine for now, and adding github auth at some point in the future is totally plausible!
